test/rules/avoid_shadowing.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_shadowing`
+
+var top = null;
```

Comment:
I don't see any tests for shadowing parameter names.

Also, I suspect that the rule as written will produce multiple lints for the same location if the name being shadowed is defined in multiple locations. Consider a test such as:
```
class C {
  int x;
  m(int x) {
    int x; // LINT
  }
}
```
(Can we verify that only one lint is produced in this case?)

Reply:
done.

It looks like several report on the same token for the same rule do not result in lint duplications.